Tests/SentryTests/Integrations/Performance/FramesTracking/SentryFramesTrackerTests.swift

```diff
@@ -114,18 +126,47 @@ private extension SentryFramesTrackerTests {
         if let frozen = frozen {
             XCTAssertEqual(frozen, currentFrames.frozen)
         }
-#if SENTRY_TARGET_PROFILING_SUPPORTED
-        if ((slow ?? 0) + (frozen ?? 0)) > 0 {
-            XCTAssertGreaterThan(currentFrames.frameTimestamps.count, 0)
-            for frame in currentFrames.frameTimestamps {
-                XCTAssertFalse(frame["start_timestamp"] == frame["end_timestamp"])
+
+#if os(iOS) || os(macOS) || targetEnvironment(macCatalyst)
+        assertProfilingData(slow: slow, frozen: frozen, frameRates: frameRates)
+#endif
+    }
+
+#if os(iOS) || os(macOS) || targetEnvironment(macCatalyst)
+    func assertProfilingData(slow: UInt? = nil, frozen: UInt? = nil, frameRates: UInt? = nil) {
+        func assertStartAndEndOrdering(frame: [String: NSNumber]) {
+            guard let start = frame["start_timestamp"] else {
+                XCTFail("Expected a start timestamp for the frame.")
+                return
+            }
```

Comment:
`l`: You can achieve similar behavior with `try XCTUnwrap(frame["start_timestamp"], "Expected a start timestamp for the frame.")`, but then you need to mark plenty of methods with throws, but sometimes it's less code.

Reply:
I keep forgetting about this one. I'm ok converting the tests to throwing, I like that model better as you've seen elsewhere 👍🏻 let's keep at that!